packages/botonic-plugin-google-translate/src/index.ts

```diff
@@ -5,27 +5,37 @@ import { AccessToken } from './access-token'
 import { GoogleTranslateApiService } from './google-translate-api-service'
 import { LanguageDetector } from './language-detector'
 import { PluginOptions } from './options'
+import { Translator } from './translator'
 
 export default class BotonicPluginGoogleTranslate implements Plugin {
-  private readonly service: GoogleTranslateApiService
+  private readonly translator: Translator
   private readonly languageDetector: LanguageDetector
 
   constructor(readonly options: PluginOptions) {
-    this.service = new GoogleTranslateApiService(
+    const service = new GoogleTranslateApiService(
       new AccessToken(this.options.credentials),
       this.options.credentials.projectId
     )
+    this.translator = new Translator(service)
     this.languageDetector = new LanguageDetector(
-      this.service,
-      this.options.whitelist || []
+      service,
+      this.options.whitelist
     )
   }
 
   async pre(request: PluginPreRequest): Promise<void> {
     try {
       if (request.input.type == INPUT.TEXT && !request.input.payload) {
         const text = request.input.data
-        request.input.language = await this.languageDetector.detect(text)
+        const translations = await this.translator.translate(
+          text,
+          this.options.translateTo
+        )
+        const detectedLanguage = await this.languageDetector.detect(text)
```

Comment:
why do we want to call detect if translation already guesses the language.
In case you need both, use Promise.all to parallelize

Reply:
We don't reuse the detected language of the translation because the language detection can return more than one language. This is useful when a whitelist is defined, and the first one is not included, but the second is.